src/include/connection.h

```diff
@@ -587,14 +587,18 @@ struct __wt_connection_impl {
     uint16_t debug_flags;
 
 /* AUTOMATIC FLAG VALUE GENERATION START 0 */
-#define WT_DIAG_ALL 0x01u
-#define WT_DIAG_CONCURRENT_ACCESS 0x02u
-#define WT_DIAG_DATA_VALIDATION 0x04u
-#define WT_DIAG_INVALID_OP 0x08u
-#define WT_DIAG_OUT_OF_ORDER 0x10u
-#define WT_DIAG_PANIC 0x20u
-#define WT_DIAG_SLOW_OPERATION 0x40u
-#define WT_DIAG_VISIBILITY 0x80u
+#define WT_DIAG_ALL 0x001u
```

Comment:
There's a few things in here I'd like some input on:
- `WT_DIAG_CURSOR_ACQUISITION` and `WT_DIAG_OBSOLETE_DATA` are category names I'm not entirely happy with. Any better recommendations are welcome.
- I'm not sure if we want to be exposing a config item named `data_loss` here. For other items like history store I've preferred the word `validation` over dc for similar reasons.

Reply:
I'm OK with the names you have chosen, but @sueloverso is my favorite naming guru. Would you mind taking a look and suggesting any changes you'd like?

Reply:
Searching, `CURSOR_ACQUISITION` is used in a single location in LSM. That could be changed to `CURSOR_CHECK`.

In general, you could consider `VALIDATE` instead of `VALIDATION` and make some of those verbs. You could consider synonyms such as `CHECK` or `CERTIFY` or `ASSURE`.

For `DATA_LOSS` you could consider `PAGE_REF_CHECK`

Reply:
Thanks @sueloverso!
We don't have anything detailed at the moment so I'll write something up.
In the meantime as a quick summary this is part of PM-2897 where we're changing some of our assertion/diagnostic code from compile-time enabled (via `HAVE_DIAGNOSTIC`) to also be runtime enabled via a connection flag. Note that existing `HAVE_DIAGNOSTIC=1` behaviour is maintained. 
The intention is when we're debugging an issue in the field we can request the customer turn on some of these checks/asserts to give us additional details and speed up investigation.

We can turn on all diagnostic checks at once, but we can also enable just a subset via the categories from this PR. This helps mitigate the performance costs of running error checking code on a production system. All code that is now runtime enable-able is in a `WT_ASSERT_OPTIONAL` or an `if(EXTRA_DIAGNOSTICS_ENABLED())` branch and is being touched by this PR.

Ideally the categories are general enough to make sense to a TSE/non-WiredTiger which is where `KEY_OUT_OF_ORDER` and `DATA_LOSS` came from. I like `CURSOR_CHECK` above but I might toy around with the `PAGE_REF_CHECK` suggestion and see if I can be clearer about what these checks failing implies.
Thanks again!

Reply:
@ajmorton thanks for the explanation. Something like PAGE_REF_CHECK is meaningful to a WT engineer but maybe less to a TSE. We'll see what you come up with.